lib/internal/Magento/Framework/Api/SearchCriteria/CollectionProcessor/FilterProcessor.php

```diff
@@ -140,7 +140,7 @@ private function checkFromTo(&$fields, &$conditions)
         $_fields = array_unique($fields);
         $_conditions = [];
         foreach ($conditions as $condition) {
-            $_conditions[array_key_first($condition)] = array_first($condition);
+            $_conditions[array_key_first($condition)] = empty($condition) ? null : reset($condition);
```

Comment:
I don't see any changes, because `array_first` contain same logic.

```
function array_first(array $array, $default = null) {
        if (empty($array)) {
            return $default;
        }

        return reset($array);
    }
```
Please correct me if I'm not right.

Reply:
Sorry, my fail. I have checked issue and see the problem